lib/Mojo/WebSocketProxy/Backend/ConsumerGroups.pm

```diff
@@ -193,22 +193,27 @@ sub call_rpc {
             foreach my $hook ($before_get_rpc_response_hooks->@*) { $hook->($c, $req_storage) }
 
             return Future->done unless $c && $c->tx;
-            my $api_response;
-            my $result;
 
-            try {
-                $result = MojoX::JSON::RPC::Client::ReturnObject->new(rpc_response => $message->{response});
-                foreach my $hook ($after_got_rpc_response_hooks->@*) { $hook->($c, $req_storage, $result) }
-                $api_response = $rpc_response_cb->($result->result);
-            }
-            catch {
-                my $error = $@;
-                $rpc_failure_cb->($c, $result, $req_storage) if $rpc_failure_cb;
-                $api_response = $c->wsp_error($msg_type, 'WrongResponse', 'Sorry, an error occurred while processing your request.');
-            };
+            my $result = MojoX::JSON::RPC::Client::ReturnObject->new(rpc_response => $message->{response});
 
-            return Future->done  if $block_response || !$api_response;
+            foreach my $hook ($after_got_rpc_response_hooks->@*) { $hook->($c, $req_storage, $result) }
 
+            my $api_response;
+            if ($result->is_error){
```

Comment:
This method doesn't work, we're using incorrect implementation of jsonrpc.

Reply:
but probably nice to have this if to be consistent with jsonrpc implementation...

Reply:
Thanks, the task not finished already and is in progress. will consider your comments.